tracer/src/Datadog.Trace/AppSec/RateLimiterTimer.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="RateLimiterTimer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.AppSec
+{
+    internal class RateLimiterTimer : IDisposable
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<RateLimiterTimer>();
+        private readonly Timer _timer;
+        private readonly int _traceRateLimit;
+        private long tracesCount;
+
+        public RateLimiterTimer(int traceRateLimit)
```

Comment:
I think it would be beneficial to have some unit tests for this - might be easier than trying to do it all via integration tests.

e.g. you can use unit tests to make sure this is really working as expected (e.g. with a couple of callbacks etc) and just use the integration tests to verify app sec is using it. That should be sufficient IMO, and might reduce any flakiness